lib/charms/traefik_k8s/v2/ingress.py

```diff
@@ -633,34 +653,53 @@ def provide_ingress_requirements(
             scheme: Scheme to be used; if unspecified, use the one used by __init__.
             host: Hostname to be used by the ingress provider to address the
              requirer unit; if unspecified, FQDN will be used instead
+            ip: Alternative addressing method other than host to be used by the ingress provider.
+                if unspecified, binding address from juju network API will be used.
             port: the port of the service (required)
         """
         for relation in self.relations:
-            self._provide_ingress_requirements(scheme, host, port, relation)
+            self._provide_ingress_requirements(scheme, host, ip, port, relation)
 
     def _provide_ingress_requirements(
         self,
         scheme: Optional[str],
         host: Optional[str],
+        ip: Optional[str],
         port: int,
         relation: Relation,
     ):
         if self.unit.is_leader():
             self._publish_app_data(scheme, port, relation)
 
-        self._publish_unit_data(host, relation)
+        self._publish_unit_data(host, ip, relation)
 
     def _publish_unit_data(
         self,
         host: Optional[str],
+        ip: Optional[str],
         relation: Relation,
     ):
         if not host:
             host = socket.getfqdn()
 
+        if not ip:
```

Comment:
I think this whole `if not ip` block can be simplified:

```suggestion
if ip is None:
    network_binding = self.charm.model.get_binding(relation)
    
    if network_binding is not None and (bind_address := network_binding.network.bind_address) is not None:
        ip = str(bind_address)
    else:
        log.error("failed to retrieve ip information from juju")
```

Reply:
Updated, thanks!